src/services/outliningElementsCollector.ts

```diff
@@ -84,6 +85,7 @@ namespace ts.OutliningElementsCollector {
         let depth = 0;
         const maxDepth = 20;
         function walk(n: Node): void {
+            cancellationToken.throwIfCancellationRequested();

```

Comment:
we already have a [ThrottledCancellationToken](https://github.com/Microsoft/TypeScript/blob/master/src/services/shims.ts#L473) in `shims.ts`, we can move to `services.ts` and use here